pilot/pkg/proxy/envoy/v2/discovery.go

```diff
@@ -340,35 +333,26 @@ func (s *DiscoveryServer) ClearCache() {
 }
 
 // Start the actual push. Called from a timer.
-func (s *DiscoveryServer) doPush(req *model.UpdateRequest) {
-	// more config update events may happen while doPush is processing.
-	// we don't want to lose updates.
-	s.mutex.Lock()
-	// Swap the edsUpdates map - tracking requests for incremental updates.
-	// The changes to the map are protected by ds.mutex.
-	edsUpdates := s.edsUpdates
```

Comment:
Nice.. much easier to understand now. Can you just get rid of doPush now and make it Push though